sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/options/ProxyInvocationHandler.java

```diff
@@ -355,10 +472,24 @@ public void serialize(PipelineOptions value, JsonGenerator jgen, SerializerProvi
         // instances that have been modified since the previous serialization.
         Map<String, Object> serializableOptions =
             Maps.<String, Object>newHashMap(handler.jsonOptions);
-        serializableOptions.putAll(filteredOptions);
+        for (Map.Entry<String, BoundValue> entry : filteredOptions.entrySet()) {
+          serializableOptions.put(entry.getKey(), entry.getValue().getValue());
+        }
+
+
         jgen.writeStartObject();
         jgen.writeFieldName("options");
         jgen.writeObject(serializableOptions);
+
+        if (handler.canPopulateDisplayData()) {
```

Comment:
More generally, this would fail if we try to serialize PipelineOptions which were previously deserialized from JSON. I don't know of actual scenarios that exercise this, but there are existing tests that do.